com.unity.ml-agents/Runtime/Actuators/ActuatorList.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Unity.MLAgents.Actuators
+{
+    public class ActuatorList : IList<IActuator>
+    {
+        float[] m_Actions;
+        IList<IActuator> m_Actuators;
+
+        public ActuatorList(int capacity = 0)
+        {
+            m_Actuators = new List<IActuator>(capacity);
+        }
+
+        public float[] StoredActions
+        {
+            get { return m_Actions; }
+        }
+
+        public void EnsureActionBufferSize()
+        {
+            var size = 0;
+            for (int i = 0; i < m_Actuators.Count; i++)
+            {
+                size += m_Actuators[i].GetNumberOfActions();
+            }
+
+            m_Actions = new float[size];
+        }
+
+        public void UpdateActions(float[] fullActionBuffer)
+        {
+            if (fullActionBuffer == null)
+            {
+                Array.Clear(m_Actions, 0, m_Actions.Length);
+            }
+            else
+            {
+                Array.Copy(fullActionBuffer, m_Actions, m_Actions.Length);
+            }
+            var start = 0;
+            for (int i = 0; i < m_Actuators.Count; i++)
+            {
+                var actuator = m_Actuators[i];
+                actuator.UpdateActions(m_Actions, start);
+                start += actuator.GetNumberOfActions();
+            }
+        }
+
+        public void ExecuteActions()
+        {
+            for (int i = 0; i < m_Actuators.Count; i++)
+            {
+                m_Actuators[i].OnActionReceived();
```

Comment:
How about making OnActionReceived() take an IList<float> or ArraySegment<float> instead? That would mean they wouldn't need to do their own bookkeeping in UpdateActions (and we could probably drop that from the interface).

Reply:
yeah, I was thinking about that as well.  I didn't want to stray too far from the original interface.  I like that approach better as well. I'll change it up.